crates/bcr-wallet-core/src/wallet/utils.rs

```diff
@@ -0,0 +1,73 @@
+// ----- standard library imports
+use std::collections::HashMap;
+// ----- extra library imports
+// ----- local modules
+// ----- end imports
+
+pub fn select_proofs_for_amount(
+    proofs: &[cashu::Proof],
+    send_amount: u64,
+) -> Option<Vec<cashu::Proof>> {
+    let mut dp: HashMap<u64, Vec<cashu::Proof>> = HashMap::new();
+    dp.insert(0, Vec::new());
+
+    for proof in proofs {
+        let current_dp = dp.clone();
+        for (sum, subset) in current_dp.iter() {
+            let new_sum = sum + u64::from(proof.amount);
+            if new_sum > send_amount {
+                continue;
+            }
+            if !dp.contains_key(&new_sum) {
+                let mut new_subset = subset.clone();
+                new_subset.push(proof.clone());
+                dp.insert(new_sum, new_subset);
+            }
+        }
+    }
+
+    dp.get(&send_amount).cloned()
+}
+
+pub fn get_amounts(mut targ: u64) -> Vec<u64> {
```

Comment:
while I very much appreciate using bitwise operators to split out the powers of two, I feel like a few comments on what this method does and why/how wouldn't hurt, since this stuff isn't used a lot in most Rust code (e.g. for external contributors)